python/private/pypi/parse_requirements.bzl

```
@@ -292,6 +292,21 @@ def _add_dists(*, requirement, index_urls, logger = None):
         index_urls: The result of simpleapi_download.
         logger: A logger for printing diagnostic info.
     """
+
+    # Handle direct URLs in requirements
+    if hasattr(requirement.srcs, "url"):
```

it would be better to just return the attribute always but leave it as empty string if it is not set.

Then the logic would read as:
```suggestion
    if requirement.srcs.url:
```